### DISCUSSION TITLE
the speed of using nested fileds

### DISCUSSION THREAD

**Discussion by Novaoceans** [Parcels-code/Parcels#1307] [Q&A]
Hi, 
When I used single NEMO fields to run ~10k particles, it usually took less than 3 minutes to finish a 20-days' simulation.
But when I used 2 NEMO fields with different resolution and area coverage, it took more than 2 hours to run ~100 particles for a 20-days' simulation. These tests are under normal OceanParcels package. Is this normal situation or is there anything wrong with this?
Thanks for your help!

**Comment by erikvansebille** [comment]
Hi @Novaoceans, thanks for asking. You are right that using `NestedFields` can be _much_ slower than using no nesting. That's because the transfer of particles from one nest to another happens outside the fast inner-`JIT`-loop in C; but instead is dealt with in the slower python loop around it. 

While we implemented `NestedFields` more than five years ago, we've never really developed it further. Not many users needed it, so there was little demand for optimizing the feature. Moving the transfer of Particles to the `JIT`-loop would take a significant amount of development time, I fear. 

Can you work around it?

**Reply by erikvansebille** [reply]
Note that I have now also added a comment about this in the `NestedFields` tutorial: https://github.com/OceanParcels/parcels/commit/2972f7f498bfc1668242d3c6f0ad764c9f60da89
